webapp/stores/team_store.jsx

```diff
@@ -321,6 +321,14 @@ class TeamStoreClass extends EventEmitter {
         }
     }
 
+    subtractUnread(teamId, msgs, mentions) {
```

Comment:
Yes, the readon for this is as following:

Lets day you have a mention in two different channels in Team A and you are viewing Team B, you switch to team A and go to one of the channels it will subtract just that mention from the team store leaving the mention count as 1.

When switching to team B again it will show just 1 mention for Team A while it gets the mention count from the server the same when your mention count goes to 0 and that will avoid the flickering thing.